test/release/examples/patterns/readcsv.chpl

```diff
@@ -0,0 +1,286 @@
+    }
+    dataRows.append(aRowMap);
+  }
+  reader.close();
+
+  // Print out the last data row to illustrate what one of the row maps contains.
+  if debug then writeln("\ndataRows[dataRows.size-1] = ", dataRows[dataRows.size-1]);
+ 
+  //-----------------------------------------------------------------------
+  // To see how the data is accessed and for testing, let's generate a 
+  // temporary csv file from the stored data.  It should be the same as the 
+  // input csv file.
+  if debug {
+    var outfile = open(approach1FileName, iomode.cw);
+    var writer = outfile.writer();
+
+    // First write to the output file the column names separated by commas.
+    for colIdx in 0..colNames.size-2 {
+      writer.write(colNames[colIdx],",");
+    }
+    // The last column name should not be followed by comma.
+    writer.writeln(colNames[colNames.size-1]);
+
+    // Then write to the file each row of data.
+    for row in dataRows {
+      for colIdx in 0..colNames.size-2 {
+        writer.write(row[colNames[colIdx]],",");
+      }
+      // The last column value should not be followed by comma.
+      writer.writeln(row[colNames[colNames.size-1]]);
+    }
+    writer.close();
+
+    // Compare the file generated from the list of maps with the original input csv file.
+    checkThatFilesMatch(inFileName,approach1FileName);
+  }
+}
+
+// Approach 2: chapeltastic -- create an associative array with the key being the 
+//             column name and the value being an array of strings
+{
+  writeln();
+  writeln();
+  writeln("Approach 2: associative array keyed of column name with array of vals per column"); 
+
+  // Create another reader of the input csv file
+  var reader = f.reader();
+
+  // Read the first line to get the column names.
+  // Note: this portion is the same as in Approach 1
+  var line : string;
+
+  if (!reader.readline(line)) then
+    writeln("ERROR: ", inFileName, " appears to be empty");
+
+  var colNames : list(string);
+  colNames = createListOfColNames(line);
+
+  if debug {
+    writeln("\ncolNames: ", colNames);
+  }
+
+  // Create a list of lines from the file, so we know how many lines of data there are.
+  // Then create an associative array with an entry per column of data, with the
+  // associative key being the column name.  The value will be an array of strings
+  // with the column's value for each row in the data set.
+  var dataRows : list(string);
+
+  // Reading all of the lines of the file into a list.
+  while (reader.readline(line)) {
+    dataRows.append(line);
+  }
+
+  // Declaring an associative array, where the value type is a 1D array large enough
+  // to store one column value per row of data.
+  var valDomain = {0..dataRows.size-1}; // Domain indicating how many values per column.
+  var colNameDomain: domain(string);    // An associative domain of strings.
+  for colName in colNames do            // Put each column name into the column name domain.
+    colNameDomain += colName;
```

Comment:
I *think* you can bulk-add stuff into associative domains. (I am sure you can do it in sparse, and it is optimized, but I expect the associative case to work but without optimization)

```chapel
colNameDomain += colNames;
```

You can even collapse everything together probably:

```chapel
var colNameDomain: domain(string) = colNames;
```

Things like this works in sparse, but I am not that sure about associative.

Reply:
done, nice.